src/ctapipe/io/hdf5monitoringsource.py

```diff
@@ -0,0 +1,593 @@
+        self._read_and_validate_subarrays()
+
+        # Process all monitoring files
+        for file in self.input_files:
+            self._process_single_file(file)
+
+    def _read_and_validate_subarrays(self):
+        """Read subarray descriptions from files and validate compatibility."""
+        # Loop over the input files to read the subarray description and check for compatibility
+        # if a subarray is already provided either externally or via a previous monitoring file.
+        subarrays = ([self.subarray] if self.subarray is not None else []) + [
+            SubarrayDescription.from_hdf(f) for f in self.input_files
+        ]
+        # Check if all subarray descriptions are compatible
+        if not SubarrayDescription.check_matching_subarrays(subarrays):
+            raise IOError("Incompatible subarray descriptions found in input files.")
+        # Set the subarray description
+        self.subarray = subarrays[0]
+
+    def _process_single_file(self, file):
+        """Process a single monitoring file."""
+        # Add the file to the provenance
+        Provenance().add_input_file(
+            str(file),
+            role="Monitoring",
+            reference_meta=read_reference_metadata(file),
+        )
+
+        with tables.open_file(file) as open_file:
+            # Validate simulation consistency
+            file_is_simulation = "simulation" in open_file.root
+
+            if self._is_simulation is None:
+                self._is_simulation = file_is_simulation
+            else:
+                if self._is_simulation != file_is_simulation:
+                    raise IOError(
+                        f"HDF5MonitoringSource: Inconsistent simulation flags found in "
+                        f"file '{file}'. Previously processed files have "
+                        f"simulation flag set to {self._is_simulation}, while "
+                        f"current file has it set to {file_is_simulation}."
+                    )
+
+            # Get monitoring types from the file
+            file_monitoring_types = get_hdf5_monitoring_types(open_file)
+            # Check for overlapping monitoring types
+            overlapping_types = set(file_monitoring_types) & self._monitoring_types
+            if overlapping_types:
+                overlapping_names = [mt.name for mt in overlapping_types]
+                msg = (
+                    f"File '{file}' contains monitoring types {overlapping_names} "
+                    f"that are already present in previously processed files. "
+                    f"This may indicate duplicate or overlapping monitoring data."
+                )
+                self.log.warning(msg)
+                warnings.warn(msg, UserWarning)
+            # Update monitoring types
+            self._monitoring_types.update(file_monitoring_types)
+
+        # Process each monitoring type
+        if MonitoringType.PIXEL_STATISTICS in file_monitoring_types:
```

Comment:
Hi @TjarkMiener,  to be sure to understand, why this call i necessary here?

Reply:
Why wouldn't it be?

Reply:
We are planning to expand this component once more monitoring types are available, e.g. optical throughput etc.